src/storage/redis_metadata.h

```diff
@@ -364,3 +365,36 @@ class TDigestMetadata : public Metadata {
 
   double Delta() const { return 1. / static_cast<double>(compression); }
 };
+
+enum class TSDuplicatePolicy : uint8_t {
+  BLOCK = 0,
+  FIRST = 1,
+  LAST = 2,
+  MIN = 3,
+  MAX = 4,
+  SUM = 5,
+};
+
+class TimeSeriesMetadata : public Metadata {
+ public:
+  enum class ChunkType : uint8_t {
+    UNCOMPRESSED = 0,
+    COMPRESSED = 1,
+  };
+
+  uint64_t retention_time;
+  uint64_t chunk_size;
+  ChunkType chunk_type;
+  TSDuplicatePolicy duplicate_policy;
+  Slice source_key;
```

Comment:
`Slice` has no ownership. It's like a `string_view`.

Here I think we can hold an owned string so that we don't need to worry about lifetime.